transformations/transformations/data_offload.py

```diff
@@ -27,13 +27,17 @@ class DataOffloadTransformation(Transformation):
     ----------
     remove_openmp : bool
         Remove any existing OpenMP pragmas inside the marked region.
+    deviceptr : bool
```

Comment:
It's a little hard to judge from the brief option name what that implies. I would propose to follow the typical "start with a verb" naming scheme, e.g., by calling this `assume_deviceptr`? That implies imho also the requirement this imposes on the calling code to handle the offload.

Reply:
Seconded, either `assume_deviceptr` or `use_deviceptr` or `mark_as_deviceptr` (or similar). And ideally propagate change up to cmake layer (explicit is better than implicit :wink: )